### PR TITLE
[2.3][Process] Fixed PhpProcess::getCommandLine() result

### PR DESCRIPTION
**Pull request by francisbesset**
The `PhpProcess::getCommandLine()` return `null` if `PhpProcess::start()` was not called.

``` php
$process = new PhpProcess(<<<PHP
<?php echo "foobar";
PHP
);

$process->getCommandLine(); // return null
$process->start();
$process->getCommandLine(); // return the PHP binary path
```

This PR fix the problem.

| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Process/Tests/PhpProcessTest.php [hunk 1]
```diff
@@ -26,4 +27,27 @@ public function testNonBlockingWorks()
         $process->wait();
         $this->assertEquals($expected, $process->getOutput());
     }
+
+    public function testCommandLine()
+    {
+        if (defined('PHP_BINARY')) {
+            $this->markTestSkipped('The PHP binary is easily available as of PHP 5.4');
```
Comment (stof) [inline src/Symfony/Component/Process/Tests/PhpProcessTest.php]: Do you really need to skip this test ?

File: src/Symfony/Component/Process/Tests/PhpProcessTest.php [hunk 2]
```diff
@@ -26,4 +27,27 @@ public function testNonBlockingWorks()
...
+        $f = new PhpExecutableFinder();
+        $commandLine = $f->find();
+
+        $this->assertSame($commandLine, $process->getCommandLine(), '::getCommandLine() returns the command line of PHP');
+
+        $process->start();
+        $this->assertSame($commandLine, $process->getCommandLine(), '::getCommandLine() returns the command line of PHP');
+
+        $process->wait();
+        $this->assertSame($commandLine, $process->getCommandLine(), '::getCommandLine() returns the command line of PHP');
```
Comment (stof) [inline src/Symfony/Component/Process/Tests/PhpProcessTest.php]: the message should not be the same 3 times. It should change to describe the difference between these checks

File: src/Symfony/Component/Process/Tests/PhpProcessTest.php (hunk 1 above)
Comment (francisbesset) [inline src/Symfony/Component/Process/Tests/PhpProcessTest.php]: Maybe not, but [ExecutableFinderTest.php](https://github.com/symfony/symfony/blob/2.3/src/Symfony/Component/Process/Tests/ExecutableFinderTest.php#L39) have this statements.

**Comment by romainneutron** [comment]
Hello and thanks for this pull request!

I'm not a big fan of a getter that modifies the inner property. Actually, I'd prefer to move the PHP binary detection in the constructor, and throw an exception in the `start` method in case `getCommandLine` returns `null`

That way, we don't change `getCommandLine` behavior and I'd be much more happy :)

Are you okay with this?

**Comment by francisbesset** [comment]
@romainneutron No problem for me. It's fixed :)

**Comment by romainneutron** [comment]
Cool :+1:

**Comment by fabpot** [comment]
Thank you @francisbesset.

**Pull request closed without merging**